sdk/storage/storage-blob/review/storage-blob.api.ts

```diff
@@ -0,0 +1,1424 @@
+  getStatistics(options?: ServiceGetStatisticsOptions): Promise<Models.ServiceGetStatisticsResponse>;
+  getUserDelegationKey(start: Date, expiry: Date, options?: ServiceGetUserDelegationKeyOptions): Promise<ServiceGetUserDelegationKeyResponse>;
+  // WARNING: The type "Models.ContainerItem" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.ServiceListContainersSegmentResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.ContainerItem" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.ServiceListContainersSegmentResponse" needs to be exported by the package (e.g. added to index.ts)
+  listContainers(options?: ServiceListContainersOptions): PagedAsyncIterableIterator<Models.ContainerItem, Models.ServiceListContainersSegmentResponse>;
+  // WARNING: The type "Models.StorageServiceProperties" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.ServiceSetPropertiesResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.StorageServiceProperties" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.ServiceSetPropertiesResponse" needs to be exported by the package (e.g. added to index.ts)
+  setProperties(properties: Models.StorageServiceProperties, options?: ServiceSetPropertiesOptions): Promise<Models.ServiceSetPropertiesResponse>;
+  submitBatch(batchRequest: BatchRequest, options?: ServiceSubmitBatchOptionalParams): Promise<ServiceSubmitBatchResponse>;
+}
+
+// @public
+interface BlobSetHTTPHeadersOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+}
+
+// @public
+interface BlobSetMetadataOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+}
+
+// @public
+interface BlobSetTierOptions {
+  abortSignal?: AbortSignalLike;
+  leaseAccessConditions?: Models.LeaseAccessConditions;
+  rehydratePriority?: Models.RehydratePriority;
+}
+
+// @public
+interface BlobStartCopyFromURLOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  metadata?: Metadata;
+  rehydratePriority?: Models.RehydratePriority;
+  sourceModifiedAccessConditions?: Models.ModifiedAccessConditions;
+  tier?: BlockBlobTier | PremiumPageBlobTier | string;
+}
+
+// @public
+interface BlobSyncCopyFromURLOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  metadata?: Metadata;
+  sourceModifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobUndeleteOptions {
+  abortSignal?: AbortSignalLike;
+  customerProvidedKey?: Models.CpkInfo;
+}
+
+// @public
+class BlockBlobClient extends BlobClient {
```

Comment:
I would like to see a method to generate a SAS URI fragment. It can throw if you constructed your client using anything other than a SharedKeyCredential.

I would also like an accessor to get the URL for a blob. That way I can generate the SAS token for a blob and simply append it to the Blob URL.

Reply:
Options:

1. Throw if credential can't generate SAS.
2. The method can take the shared key.
3. BlockBlobClient could generate the params for generateSAS.

Consider more offline

Reply:
BlobClient SAS generation could also include the full URL.

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/5241